apps/air-discount-scheme/api/src/app/modules/discount/discount.resolver.ts

```diff
@@ -72,7 +69,10 @@ export class DiscountResolver {
           if (!discount || discount.expiresIn <= TWO_HOURS) {
             discount = await backendApi.createDiscount(relation.nationalId)
           }
-          return [...acc, { ...discount, user: relation }]
+          return [
+            ...acc,
+            { ...discount, user: { ...relation, fund: discount.user.fund } },
```

Comment:
The user fund is recalculated in `getUserRelations`.
The explicit discount credit bypasses the validation and is different than what is calculated for a normal user.
Therefore we are overwriting the generated user fund with the cached user fund but they differ because of this line:
https://github.com/island-is/island.is/blob/46a658ba37ff0a3c4c2c7aea4ff55ef9acbe72ba/apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts#L96

From `getRelations -> getMultipleUsersByNationalIdArray -> getUserByNationalId`